### PR TITLE
chore: add configuration to force websocket  (WPB-3311)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3311" title="WPB-3311" target="_blank">TaskWPB-3311</a>  Turn on Websockets by default via config flag
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We need to provide a way to enable websocket by default using configuration.

### Solutions

Add additional condition to check for custom-build config => ` "websocket_enabled_by_default": true,`

### Dependencies (Optional)

https://github.com/wireapp/android-customization

Needs releases with:

- [x] GitHub link to other pull request

#### How to Test

After merge of https://github.com/wireapp/android-customization we can trigger a build using this branch 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-2018/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.5(7150436)(7150436)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.5(7150436)(7150436)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/di/KaliumConfigsModule.kt [hunk 1]
```diff
@@ -69,7 +69,7 @@ class KaliumConfigsModule {
             wipeOnCookieInvalid = BuildConfig.WIPE_ON_COOKIE_INVALID,
             wipeOnDeviceRemoval = BuildConfig.WIPE_ON_DEVICE_REMOVAL,
             wipeOnRootedDevice = BuildConfig.WIPE_ON_ROOTED_DEVICE,
-            isWebSocketEnabledByDefault = !context.isGoogleServicesAvailable()
+            isWebSocketEnabledByDefault = BuildConfig.WEBSOCKET_ENABLED_BY_DEFAULT || !context.isGoogleServicesAvailable()
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/di/KaliumConfigsModule.kt]: can we remove the !context.isGoogleServicesAvailable() ?

File: app/src/main/kotlin/com/wire/android/ui/home/settings/appsettings/networkSettings/NetworkSettingsScreen.kt [hunk 2]
```diff
@@ -84,7 +85,7 @@ fun NetworkSettingsScreenContent(
 @Composable
 private fun getSwitchState(isWebSocketEnabled: Boolean, setWebSocketState: (Boolean) -> Unit): SwitchState {
     val context = LocalContext.current
-    return if (context.isGoogleServicesAvailable()) {
+    return if (BuildConfig.WEBSOCKET_ENABLED_BY_DEFAULT || context.isGoogleServicesAvailable()) {
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/home/settings/appsettings/networkSettings/NetworkSettingsScreen.kt]: this line is duplicated can we extract it to its own usecase ?

File: app/src/main/kotlin/com/wire/android/di/KaliumConfigsModule.kt (hunk 1 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/di/KaliumConfigsModule.kt]: Is part of the requirement, we need to enable it by default if no Google play services available.
But the `BuildConfig` has preference if true

File: app/src/main/kotlin/com/wire/android/ui/home/settings/appsettings/networkSettings/NetworkSettingsScreen.kt (hunk 2 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/ui/home/settings/appsettings/networkSettings/NetworkSettingsScreen.kt]: I'll try to create something for future reusage, but the check is not the same actually
edit: I moved it to a common function, given now it's the same logic =)

**Review by tmspzz — APPROVED** [review]

**Comment by MohamadJaara** [comment]
@yamilmedina  There is a switch to force web socket connection, can we set it to on and disable it when it is enforced?

to avoid any issues if someone tried to switch it on when it is enforced

**Comment by yamilmedina** [comment]
> @yamilmedina There is a switch to force web socket connection, can we set it to on and disable it when it is enforced?
> 
> to avoid any issues if someone tried to switch it on when it is enforced

After several days/weeks with input from customer and back and forth, we reached an agreement, here is what's implemented in detail.

https://wearezeta.atlassian.net/wiki/spaces/~751534962/pages/855670923/20230726+-+A+Websockets+tale

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-2018/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.5(7159168)(7159168)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.5(7159168)(7159168)-staging-compat.apk)

**Review by vitorhugods — APPROVED** [review]

**Pull request merged**